prime-router/src/main/kotlin/history/db/DeliveryHistoryDatabaseAccess.kt

```diff
@@ -266,11 +264,37 @@ class DeliveryHistoryDatabaseAccess(
             .where(whereClause).asTable(DeliveryHistoryTable.DELIVERY_HISTORY)
 
         val results = db.transactReturning { txn ->
-            search.fetchResults(
+
+            val historyResults = search.fetchResults(
                 DSL.using(txn),
                 deliveriesExpression.asterisk(),
                 deliveriesExpression.asTable(DeliveryHistoryTable.DELIVERY_HISTORY.name),
             )
+            val reportFiles = db
+                .fetchReportFileByIds(historyResults.results.map { UUID.fromString(it.reportId) })
+            historyResults.map { history ->
```

Comment:
Maybe a comment to describe what you are doing here since it may not be clear initially.